addons/binding/org.openhab.binding.tivo/src/main/java/org/openhab/binding/tivo/handler/TiVoHandler.java

```diff
@@ -0,0 +1,794 @@
+ * http://www.eclipse.org/legal/epl-v10.html
+ */
+package org.openhab.binding.tivo.handler;
+
+import static org.openhab.binding.tivo.TiVoBindingConstants.*;
+
+import java.util.Arrays;
+import java.util.List;
+import java.util.SortedSet;
+import java.util.TreeSet;
+import java.util.concurrent.ScheduledFuture;
+import java.util.concurrent.TimeUnit;
+
+import org.eclipse.smarthome.config.core.Configuration;
+import org.eclipse.smarthome.core.library.types.DecimalType;
+import org.eclipse.smarthome.core.library.types.StringType;
+import org.eclipse.smarthome.core.thing.ChannelUID;
+import org.eclipse.smarthome.core.thing.Thing;
+import org.eclipse.smarthome.core.thing.ThingStatus;
+import org.eclipse.smarthome.core.thing.ThingStatusDetail;
+import org.eclipse.smarthome.core.thing.binding.BaseThingHandler;
+import org.eclipse.smarthome.core.types.Command;
+import org.eclipse.smarthome.core.types.RefreshType;
+import org.openhab.binding.tivo.internal.service.TivoConfigData;
+import org.openhab.binding.tivo.internal.service.TivoStatusData;
+import org.openhab.binding.tivo.internal.service.TivoStatusProvider;
+import org.slf4j.Logger;
+import org.slf4j.LoggerFactory;
+
+/**
+ * The {@link TiVoHandler} is the BaseThingHandler responsible for handling commands that are
+ * sent to one of the Tivo's channels.
+ *
+ * @author Jayson Kubilis - Initial contribution
+ * @author Andrew Black (AndyXMB) - Updates / compilation corrections. Addition of channel scanning functionality.
+ */
+
+public class TiVoHandler extends BaseThingHandler {
+    private Logger logger = LoggerFactory.getLogger(TiVoHandler.class);
+    private TivoConfigData tivoCfgData = null;
+    private TivoStatusProvider myTivoService = null;
+    private ScheduledFuture<?> refreshJob;
+    private ScheduledFuture<?> chScanJob;
+
+    /**
+     * Instantiates a new TiVo handler.
+     *
+     * @param thing the thing
+     * @param discoveryServiceRegistry the discovery service registry
+     */
+    public TiVoHandler(Thing thing) {
+        super(thing);
+        logger.debug("TiVoHandler '{}' - creating", getThing().getUID());
+    }
+
+    @Override
+    public void handleCommand(ChannelUID channelUID, Command command) {
+
+        // Handles the commands from the various TiVo channel objects
+        logger.debug("handleCommand {}, command: {}", channelUID, command);
+
+        if (command != null && myTivoService != null) {
+
+            TivoStatusData myTivo = myTivoService.getServiceStatus();
+            String tivoCommand = null;
+
+            if (!isInitialized()) {
+                logger.debug("handleCommand '{}' device is not intialised yet, command '{}' will be ignored.",
+                        getThing().getUID(), channelUID + " " + command);
+                return;
+            }
+
+            // Check to see if we are running a channel scan, if so 'disable' UI commands, else chaos ensues...
+            if (myTivo != null && myTivo.getChScan()) {
```

Comment:
Please use more characters `getChScan` => `isChannelScanInProgress` then you don't need the command just the actual code.